hydromt/workflows/grid.py

```diff
@@ -110,16 +113,25 @@ def create_grid_from_region(
 
     kind = next(iter(region))
     if kind in ["bbox", "geom"]:
-        if not res:
+        if not isinstance(res, (int, float)):
```

Comment:
What is the difference  to check if it is int or float? The spec says it's an optional float, so it can be either float or None. Can you elaborate on why this check is better?

Reply:
The spec says but not type check is actually done by the spec no?

Reply:
Correct, but are you expecting someone to put a `string` there? Otherwise we have to check all incoming arguments all the time.

`res` is an `optional[float]`. That's what we expect, and we want to know if it's not `None`. If you expect an incoming `int`, then we should change the argument spec to `Optional[Union[int,float]]`. But in runtime I would only check `if not None`.